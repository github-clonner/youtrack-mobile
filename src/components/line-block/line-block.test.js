import React from 'react';
import {Text} from 'react-native';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import LineBlock from './line-block';

describe('<LineBlock/>', () => {

  let wrapper;
  let children;
  let lineBlockChildrenMock;

  beforeEach(() => {
    lineBlockChildrenMock = 'lineBlockChildrenMock';
    children = <Text testID={lineBlockChildrenMock}>'content'</Text>;
    wrapper = doShallow(children);
  });


  describe('Render', () => {
    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component', () => {
      expect(findByTestId('lineBlock')).toHaveLength(1);
    });

    it('should children', () => {
      expect(findByTestId(lineBlockChildrenMock)).toHaveLength(1);
    });

    it('should show all button', () => {
      expect(findByTestId('lineBlockMore')).toHaveLength(1);
    });


    describe('Modal', () => {

      it('should show all children', () => {
        findByTestId('lineBlockMore').simulate('press');

        expect(findByTestId(lineBlockChildrenMock)).toHaveLength(3);
      });

      it('should close modal', () => {
        findByTestId('lineBlockMore').simulate('press');
        expect(findByTestId('lineBlockModalClose')).toHaveLength(1);

        findByTestId('lineBlockModalClose').simulate('press');
        expect(findByTestId('lineBlockModal')).toHaveLength(0);
        expect(findByTestId(lineBlockChildrenMock)).toHaveLength(1);
      });
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(children: React$Element<Text>, inline: boolean) {
    return shallow(
      <LineBlock
        inline={inline}
        childrenRenderer={() => children}
        allChildrenRenderer={() => <Text>{children}{children}</Text>}
      />
    );
  }
});
