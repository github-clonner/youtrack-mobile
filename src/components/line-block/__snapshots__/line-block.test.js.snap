// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineBlock/> Render should match a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#DFE5EB",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-start",
        "marginRight": -16,
        "marginTop": 16,
        "paddingBottom": 8,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="lineBlock"
>
  <Text
    testID="lineBlockChildrenMock"
  >
    'content'
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 40,
      }
    }
    testID="lineBlockMore"
  >
    <Text
      style={
        Object {
          "lineHeight": 22,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#CCC"
        name="angle-right"
        size={24}
      />
    </Text>
  </TouchableOpacity>
</View>
`;
